src/hooks/useIsFlashbotsProtectRPC.ts

```diff
@@ -0,0 +1,42 @@
+import { Contract } from '@ethersproject/contracts'
+import { useEffect, useState } from 'react'
+
+import { useBlockNumber } from '../state/application/hooks'
+import { useActiveWeb3React } from './web3'
+
+export default function useIsFlashbotsProtectRPC(): [boolean, () => Promise<boolean>] {
```

Comment:
you should use our `useSingleCallResult` hook

Reply:
Is infura used to make the calls through the multicall hooks? I had tried using `useSingleCallResult` previously but the result from the call was always false. In order for the call to return true the call needs to be made through MetaMask while the Flashbots Protect RPC is in use.

Reply:
The RPC intercepts `FlashbotsRPCContract.isFlashRPC()` calls and returns `true` when calls through another RPC would return `false`. Since the multicall pattern batches these calls and routes them through a multicall contract the RPC cannot detect this call and intercept it.